### PR TITLE
Adds docs for watcher-c

### PR DESCRIPTION
**Pull request by AlliBalliBaba**
I've added some docs for compiling and using the library in C, as mentioned in #52. It might help someone else in the future, as I think this is a great library 👍

### CODE REVIEW

File: watcher-c/readme.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+The watcher can also be used from C by including the `watcher-c.h` file on compilation and in your project.
+
+You can cross compile the library via the `watcher-c/cross-compile.sh` script or download the precompiled library from the releases.
+
+### Compiling the library from source
+
+If you want to compile the library yourself you can follow the following example:
+
+First check out the `next` branch, compile a shared `libwatcher-c.so` library with `gcc` and install
```
Comment (e-dant) [inline watcher-c/readme.md]: This will be the main branch at some point soon

File: watcher-c/readme.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+it in the `/usr/local/lib` directory:
+
+```bash
```
Comment (e-dant) [inline watcher-c/readme.md:16]: These instructions, relying on gcc and manual copying into a binary path, seem tailored to container-based environments. 

Meson might be a good mention here. 

I'm not opposed to naming a compiler directly, but I think at least we should point to c++ (and let the system decide what that means) and make explicit that the copy to /usr/local/bin is both Unix specific and optional

File: watcher-c/readme.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+curl -L https://github.com/e-dant/watcher/archive/refs/heads/next.tar.gz | tar xz
+cd watcher-next/watcher-c
+gcc -o libwatcher-c.so ./src/watcher-c.cpp -I ./include -I ../include -std=c++17 -O3 -Wall -Wextra -fPIC -shared
```
Comment (e-dant) [inline watcher-c/readme.md]: I hesitate to add compiler optimizations in example instructions.

We can either be
- extremely aggressive in our optimizations (like some of our build systems/scripts do; specifying no RTTI, exceptions, asking for whole program analysis, etc),
- or very example-focused and minimal (omitting O3, for example)

Same applies to Wall, Wextra -- for example commands

---

Up to you which path we want to go down, but I would lean towards a minimal example (which is almost always good enough, anyway) by omitting optimizations and extra stuff

File: watcher-c/readme.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+After installing the library, copy the `watcher-c.h` file to your project and include it in your C code:
+
+```c
+#include "watcher-c.h"
+```
+
+You will then be able to use the `wtr_watcher_open` and `wtr_watcher_close` functions by linking against the library (`-lwatcher-c`).
+
+`wtr_watcher_close` will always return `false` if the watcher is closed before the first `@live` event has been sent.
```
Comment (e-dant) [inline watcher-c/readme.md]: I think this deserves a mention in the main readme, not the watcher-c specific readme. 

I think the right place for this is in the library usage section, right after we explain the special watcher events and lay out the important functions.

**Review by e-dant — CHANGES_REQUESTED** [review]

**Comment by e-dant** [comment]
Thank you :)

File: watcher-c/readme.md (hunk 4 above)
Comment (e-dant) [reply watcher-c/readme.md]: I added this in: https://github.com/e-dant/watcher/commit/fb9c5254f54726160dbe5a408cd6f2fae2cb8b4f -- think we can merge or remove this line?

**Comment by e-dant** [comment]
I added a little blurb for this PR in the changelog: https://github.com/e-dant/watcher/commit/d106c99dde43ed9e08996ff8643bed0927604f4f

File: watcher-c/readme.md (hunk 1 above)
Comment (AlliBalliBaba) [reply watcher-c/readme.md]: Good point, I changed it to mention that you can checkout any branch or release branch

File: watcher-c/readme.md (hunk 3 above)
Comment (AlliBalliBaba) [reply watcher-c/readme.md]: 👍  I removed `Wall` `Wextra` `O3` and changed it to `c++`

File: watcher-c/readme.md (hunk 2 above)
Comment (AlliBalliBaba) [reply watcher-c/readme.md:16]: I changed it to mention meson, and that the copying is an instruction for container based environments

File: watcher-c/readme.md (hunk 4 above)
Comment (AlliBalliBaba) [reply watcher-c/readme.md]: Line is removed and I added a link 👍

**Pull request merged**